torch/csrc/jit/codegen/cuda/lower_allocation.cpp

```diff
@@ -459,6 +460,13 @@ class AllocationInserter : public kir::MutableIrVisitor {
       alloc_dims.push_back(ir_builder.create<kir::Int>(1));
     }
 
+    // This factor is needed for data repetition in Volta mma ops.
+    auto mma_reg_factor =
```

Comment:
Having a magic factor on an allocation makes me a bit nervous. Could we change the original tiling so this factor comes out of the tiling rather than do this?

Reply:
Yes this didn't look great to me either and this magic factor is pretty much exclusively for `mma.m8n8k4` on sm70. 
The reason for this factor is because it takes data repetition (two threads holding same data) to make a straightforward and meaningful full warp matmul with this instruction, independent of how we do the tiling. 

To the first order I don't think we'd need to generalize this to anything else. 

There might be some other ways we can re-design the mma macro to avoid this (diagonal tiling?) but it'd come with other complexity costs. 

I can rename and comment more to make this very specific and I felt that it's not too terrible if we can package the magic bits well for sm70. There's another sm70 exclusive bit in lower_mma pass for `lowerMmaOperandSmemRead` and that's hopefully everything special we need to do for it.